### PR TITLE
Notes & Text Editor

### PR DESCRIPTION
**Pull request by horsemaker**
- Add Notes
- Add Masonry Layout for Notes
- Refactored Rich Text Editor

### CODE REVIEW

**Review by narharikale — APPROVED** [review]
great work yash ✨

File: src/frontend/components/ColorPalette/ColorPalette.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+const colors = [
+  "color-note-bg",
+  "color-note-1",
+  "color-note-2",
+  "color-note-3",
+  "color-note-4",
+  "color-note-5",
+  "color-note-6",
+  "color-note-7",
+];
```
Comment (t007rushi) [inline src/frontend/components/ColorPalette/ColorPalette.js:14]: Liked this approach

File: src/frontend/components/MasonryLayout/MasonryLayout.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import React from "react";
+import Masonry from "react-masonry-css";
```
Comment (t007rushi) [inline src/frontend/components/MasonryLayout/MasonryLayout.js:2]: learn something new

File: src/frontend/contexts/notes-context.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+import { SET_NOTES } from "../constants";
+import { notesReducer } from "../reducers";
+import { getNotesService } from "../services";
+import { useAuth } from "./auth-context";
+
+const NotesContext = createContext();
+
+const NotesProvider = ({ children }) => {
+  const [notes, dispatchNotes] = useReducer(notesReducer, []);
+  console.log("notes", notes);
```
Comment (t007rushi) [inline src/frontend/contexts/notes-context.js]: console here

**Review by t007rushi — APPROVED** [review]
AWESOME UI 🔥

File: src/frontend/contexts/notes-context.js (hunk 3 above)
Comment (horsemaker) [reply src/frontend/contexts/notes-context.js]: right

**Pull request merged**